Repository/PhpcrOdmRepository.php

```diff
@@ -116,4 +121,62 @@ protected function buildCollection(array $documents)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'The root directory cannot be created.');
+        Assert::startsWith($path, '/', 'The target path %s is not absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentNode = NodeHelper::createPath($this->getManager()->getPhpcrSession(), $resolvedPath);
+        if (!$parentNode instanceof NodeInterface) {
+            throw new InvalidArgumentException('No parent node created for '.$path);
+        }
+
+        if ($resource instanceof ArrayResourceCollection) {
```

Comment:
`class ArrayResourceCollection implements IteratorAggregate, ResourceCollection` you are right, but with the simpliefied solution we just look for IteratorAggregate now.